webplugin/js/app/kommunicate.js

```diff
@@ -116,6 +116,48 @@ $applozic.extend(true, Kommunicate, {
             }, SET_TIMEOUT_DURATION);
         }
     },
+    updateConversationMetadata: function (conversationMetadata) {
+        if (conversationMetadata) {
+            var conversationInfo = JSON.parse(
```

Comment:
what's the purpose of stringify and then parsing again?

Reply:
So the purpose is to clone the object as this method is supported in ES5, and we are giving support to IE

Reply:
why do we need clone and not the original object itself?

Reply:
the incoming parameter is not behaving like an object so I am cloning it to pass the object verification.

Reply:
why is the incoming parameter not behaving like an object?

Reply:
is that okay if we pass JSONstring to the function `Kommunicate.updateConversationMetadata(JSON.stringify(object))` then parse it and use object further?

Reply:
i am not able to understand why a object passed from one place, can't be used at other place. please explain it with example